sycl/test/basic_tests/spec_const_types.cpp

```diff
@@ -82,11 +82,11 @@ int main() {
 
   cl::sycl::ONEAPI::experimental::spec_constant<float, MyFloatConst> f32 =
       program.set_spec_constant<MyFloatConst>((float)get_value());
-  // CHECK-DAG: _ZTS12MyFloatConst=1|9
+  // CHECK-DAG: _ZTS12MyFloatConst=2|
 
   cl::sycl::ONEAPI::experimental::spec_constant<double, MyDoubleConst> f64 =
       program.set_spec_constant<MyDoubleConst>((double)get_value());
-  // CHECK-DAG: _ZTS13MyDoubleConst=1|10
+  // CHECK-DAG: _ZTS13MyDoubleConst=2|
```

Comment:
what happened to the second number after '|'? I recall it was spec ID + spec type

Reply:
This patch unifies handling of scalar and composite specialization constants by re-using composite spec constants representation for scalars. This means that even for a scalar spec constant we store a tuple `(ID, 0, Size)` - under the hood this is stored as base64-encoded byte array - I've decided not to include that base64-encoded string into the test, because it is not really human-readable and I'm not sure that it introduces a lot of value.

I think we need to invent some way of testing those byte-array properties though, but I would prefer to improve that as a separate PR